crates/rpc/src/v05/method/get_transaction_status.rs

```diff
@@ -0,0 +1,298 @@
+use anyhow::Context;
+use pathfinder_common::TransactionHash;
+use serde_with::skip_serializing_none;
+use starknet_gateway_types::pending::PendingData;
+
+use crate::context::RpcContext;
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct GetTransactionStatusInput {
+    transaction_hash: TransactionHash,
+}
+
+#[derive(serde::Serialize, Debug, PartialEq, Eq)]
+#[skip_serializing_none]
+pub struct GetTransactionStatusOutput {
+    finality_status: FinalityStatus,
+    execution_status: Option<ExecutionStatus>,
+}
```

Comment:
```suggestion
pub struct GetTransactionStatusOutput {
    finality_status: FinalityStatus,
    /// Not present for received or rejected transactions.
    execution_status: Option<ExecutionStatus>,
}
```

Reply:
We can also consider making this type safe by flattening into an enum and manually doing serde.
```rust
enum Output {
    Rejected,
    Received,
    AcceptedOnL1(FinalityStatus),
    AcceptedOnL2(FinalityStatus),
}
```